### PR TITLE
Add Basic DOTD Script Self-Check

### PR DESCRIPTION
**Pull request by Hamms**
Adds a basic self-check to the Dev of the Day script, primarily intended for use by devs running the script on their system for the first time to confirm that their local environment is configured correctly to enable all necessary script functionality.

I also updated the documentation at https://docs.google.com/document/d/166j9G-Kn_PeHoV4RRX9l-x-42NlahPn0sT6tTFTX5LY/edit# to recommend this check to new Devs of the Day

## Testing story

Tested locally:

```
Your Choice: A, D, H, C, S, DTS, DTT, G, R, K, DTP, L, DTL or Q?  C
Checking access to internal servers
successfully connected to staging
Connection to ip-10-0-128-226.ec2.internal closed.
Connection to gateway.code.org closed.
successfully connected to test
Connection to ip-10-0-0-254.ec2.internal closed.
Connection to gateway.code.org closed.
successfully connected to levelbuilder-staging
Connection to ip-10-0-0-250.ec2.internal closed.
Connection to gateway.code.org closed.
successfully connected to production-console
Connection to ip-10-0-1-51.ec2.internal closed.
Connection to gateway.code.org closed.
Checking access to GitHub
latest staging commit is: 0f76173d
Checking access to Slack
current Developer of the Day is: elijah
Checking access to Honeybadger
TODO
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/dotd [hunk 1]
```diff
@@ -465,6 +467,24 @@ def get_honeybadger_issues
...
+  end
+
+  puts "Checking access to GitHub"
+  puts "latest staging commit is: #{GitHub.sha('staging', true)}"
+
+  puts "Checking access to Slack"
+  puts "current Developer of the Day is: #{DevelopersTopic.dotd}"
+
+  puts "Checking access to Honeybadger"
+  puts "TODO"
```
Comment (Hamms) [inline bin/dotd]: I'm not sure what we want out of such a check, or if we even really need it. The options that come to me here are:

1. Invoke `Honeybadger.get_recent_issues`; this confirms we can perform _all_ desired functionality, but also takes some time and generates a lot of output
2. Create a more-targeted method on the `Honeybadger` class we can invoke to just check connectivity without also downloading unnecessary data
3. Don't bother to check Honeybadger access at all in this self-check

I'm leaning toward 3, open to 2, and could be convinced on 1. Open to suggestions!

Comment (hannahbergam) [reply bin/dotd]: Fwiw, I think honeybadger feels less urgent than the other checks, so would be happy with 3!

**Review by hannahbergam — APPROVED** [review]
Thank you so much for writing this!! I'm not entirely clear on how this works via gateway but what I see here makes logical sense and I trust would forge that connection between GH and gateway that we're looking for!

File: bin/dotd (hunk 1 above)
Comment (cat5inthecradle) [reply bin/dotd]: I'm fine with 3, perhaps with a note similar to your comment?

**Review by cat5inthecradle — APPROVED** [review]

**Comment by Hamms** [comment]
The `run_on` method that I use in this check is what this script uses internally for connecting via gateway: https://github.com/code-dot-org/code-dot-org/blob/00e6ddc985642c14adbcf04b1759a7a485954e87/bin/dotd#L396-L403

Specifically, your environment the other day was failing when it tried to invoke `create_branch_from_commit`, which internally uses `run_on`: https://github.com/code-dot-org/code-dot-org/blob/00e6ddc985642c14adbcf04b1759a7a485954e87/bin/dotd#L472-L485

**Review by megcrenshaw — APPROVED** [review]
I messed up my ssh config to check, and it looks good! Thanks for doing this, and for adding documentation to the DOTD doc.

**Pull request merged**